sdk/src/beta9/abstractions/taskqueue.py

```diff
@@ -230,26 +231,35 @@ def _handle_serve_interrupt(self) -> None:
         os._exit(0)  # kills all threads immediately
 
     def _serve(self, *, dir: str, object_id: str):
-        with ThreadPoolExecutor() as pool:
-            pool.submit(self.parent.sync_dir_to_workspace, dir=dir, object_id=object_id)
-
-            def _run_serve():
-                r: Optional[StartTaskQueueServeResponse] = None
-                for r in self.parent.taskqueue_stub.start_task_queue_serve(
-                    StartTaskQueueServeRequest(
-                        stub_id=self.parent.stub_id,
-                    )
-                ):
-                    if r.output != "":
-                        terminal.detail(r.output, end="")
-
-                    if r.done or r.exit_code != 0:
-                        break
-
-                if r is None or not r.done or r.exit_code != 0:
-                    terminal.error("Serve container failed ❌")
-
-            pool.submit(_run_serve)
+        def notify(*_, **__):
+            self.parent.taskqueue_stub.task_queue_serve_keep_alive(
+                TaskQueueServeKeepAliveRequest(
+                    stub_id=self.parent.stub_id,
+                )
+            )
+
+        threading.Thread(
+            target=self.parent.sync_dir_to_workspace,
+            kwargs={"dir": dir, "object_id": object_id, "on_event": notify},
+            daemon=True,
```

Comment:
This field is required so that the thread automatically shuts down after rest of the process completes